sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RemoveRedundantAggregates.scala

```diff
@@ -47,6 +47,10 @@ object RemoveRedundantAggregates extends Rule[LogicalPlan] with AliasHelper {
       } else {
         newAggregate
       }
+
+     case agg @ Aggregate(groupingExps, _, child) if agg.groupOnly && child.deterministic &&
```

Comment:
does `child.deterministic` matter here?

Reply:
We had a test like this before:
https://github.com/apache/spark/blob/a34e2afb7538894f9a264b894924e848cd6979e0/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RemoveRedundantAggregatesSuite.scala#L104-L114

Reply:
which means `child.deterministic` doesn't matter? The test you posted did optimize out one aggregate.

Reply:
Sorry. This test:
https://github.com/apache/spark/blob/f0db8ecb810b446a48a870e55b2394d3202ae8b7/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/RemoveRedundantAggregatesSuite.scala#L164-L171